planner/nodejs_planner.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+package planner
+
+import (
+	"fmt"
+	"path/filepath"
+)
+
+type NodeJsPlanner struct{}
+
+// NodeJsPlanner implements interface Planner (compile-time check)
+var _ Planner = (*NodeJsPlanner)(nil)
+
+func (n *NodeJsPlanner) Name() string {
+	return "NodeJsPlanner"
+}
+
+func (n *NodeJsPlanner) IsRelevant(srcDir string) bool {
+	packageJSONPath := filepath.Join(srcDir, "package.json")
+	return fileExists(packageJSONPath)
+}
+
+func (n *NodeJsPlanner) Plan(srcDir string) *BuildPlan {
+	pkgManager := "npm"
+	installFiles := []string{"package.json"}
+
+	npmPkgLockPath := filepath.Join(srcDir, "package-lock.json")
+	if fileExists(npmPkgLockPath) {
+		installFiles = []string{"package.json", "package-lock.json"}
+	}
+
+	yarnPkgLockPath := filepath.Join(srcDir, "yarn.lock")
+	if fileExists(yarnPkgLockPath) {
+		pkgManager = "yarn"
+		installFiles = []string{"package.json", "yarn.lock"}
+	}
+
+	return &BuildPlan{
+		Packages:       []string{},   // This should be what users set in nix packages. Leaving it empty for now.
+		InstallFiles:   installFiles, // TODO: add this to Dockerfile template.
+		InstallCommand: fmt.Sprintf("%s install", pkgManager),
+		// This does not need to be npm build or yarn build, since build script is set by users.
+		// For now we assume users will set the build script as "build", but it could be really anything.
+		// In fact, some apps don't even require a build step.
+
+		// Lucille's controversial opinion: For NodeJS projects, we should require BuildCommand and StartCommand be set by users.
+		BuildCommand: fmt.Sprintf("%s build", pkgManager),
```

Comment:
`npm build` is not valid (at least for npm@8).

Reply:
Yeah so for `build` to be valid, it has to be a command defined in your `package.json` - Hence why I don't feel comfortable making any assumptions 😄